csrc/preseg_passes/insert_reshardings.cpp

```diff
@@ -49,7 +49,11 @@ void insertReshardingsBefore(Fusion* fusion) {
       continue;
     }
 
-    auto output = expr->outputs().at(0)->as<TensorView>();
+    if (!expr->output(0)->isA<TensorView>()) {
+      continue;
+    }
+    auto output = expr->output(0)->as<TensorView>();
```

Comment:
An Expr can produce scalar outputs. Since we run InsertReshardings unconditionally, many existing tests would fail without this change.

Reply:
I'll try to separate changes in this file to another PR with tests. Will keep you posted...